internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -137,16 +138,26 @@ func (r *LocalClusterImportReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
-	hasManagedCluster, err := r.hasLocalManagedCluster(ctx)
+	clusterVersion := &configv1.ClusterVersion{}
+	namespacedName := types.NamespacedName{
+		Name: "version",
+	}
+	err := r.client.Get(ctx, namespacedName, clusterVersion)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "unable to fetch local cluster version")
+	}
+
+	localClusterName, err := r.getLocalManagedClusterName(ctx, clusterVersion)
 	if err != nil {
 		err = r.setReconciliationStatus(ctx, false, aiv1beta1.ReasonUnableToDetermineLocalClusterManagedStatus, err.Error())
 		if err != nil {
 			return ctrl.Result{}, errors.Wrap(err, "Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
 		}
-		return ctrl.Result{}, errors.Wrap(err, "error while attempting to determine presence of managed cluster")
+		return ctrl.Result{}, errors.Wrap(err, "error while attempting to determine local cluster name")
 	}
-	if hasManagedCluster {
-		if err = r.importLocalCluster(ctx, instance); err != nil {
+	if len(localClusterName) > 0 {
```

Comment:
I suppose this works, but I think the convention is to compare directly to `""` rather than check the length of the string.

My immediate reaction to seeing this was to check why you were returning a slice type from `getLocalManagedClusterName`